### ISSUE TITLE
[Tracking / LOW] [Wave Collect] [HybridApp] Investigate Native Share for HybridApp

### ISSUE THREAD

**Issue by grgia** [Expensify/App#36613]
## Background: 

V0 (NewDot-only design) [tracking issue](https://github.com/Expensify/App/issues/15603) / [design doc](https://docs.google.com/document/d/14C1VifxvIXeyLAf0XstbC-wvyAb5SQZ0AOw0-jLRyZs/edit). 

## Moving Forward: 

We originally designed this new feature for NewDot only, but as we move towards one single app (Hybrid App) on the Old Expensify App, we need to leverage/extend the existing share-to-scan receipt feature to support common chat-based sharing.

This means that in addition to Share to Scan receipt, a user will be able to share attachments/images/text/links to chats in New Expensify.

We may need to HOLD this project as HybridApp plans change or solidify, but for now we can use this issue to share results from investigations

<details><summary>Issue Owner</summary>Current Issue Owner: @grgia</details>

**Comment by BrtqKr** [comment]
Hey @grgia, I'm from Software Mansion and I'll be handling this issue

**Comment by grgia** [comment]
@BrtqKr could you post a quick update to this issue for tracking?

**Comment by BrtqKr** [comment]
The screens for this feature are mostly done, we're connecting them with onyx. Right now we're working on reusing the existing logic from the native side and expanding it with the share extension serving as the "router", which would point to the proper OldDot/NewDot flow without affecting the old logic. We're still waiting for the ios setup for AppGroups, which is already done for the old dot and is required if we want to keep everything consistent.

**Comment by grgia** [comment]
I need to review the doc for this one. Could you link here as well @BrtqKr ? I want to unblock this this week :)

**Comment by grgia** [comment]
Was deprioritized, but we should get the HL updated and get this wrapped up in the next release cycle

**Comment by LLPeckham** [comment]
Just a note - there are some blank sections in the detailed portion, so I wasn't sure if I should proceed with a detailed review.

**Comment by grgia** [comment]
> Just a note - there are some blank sections in the detailed portion, so I wasn't sure if I should proceed with a detailed review.

@LLPeckham great shout, I'll take a look at those sections tomorrow (Thurs)

**Comment by AndrewGable** [comment]
I have read and reviewed this Design Doc!

**Comment by nkuoch** [comment]
I have read and reviewed this Design Doc!

**Comment by JmillsExpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by MarcFillari** [comment]
@grgia it looks like this project was placed on HOLD in our internal tracking sheets based on this comment back in Feb https://github.com/Expensify/App/issues/15603#issuecomment-1934811394

Could you please follow this SO to have our CapSW tracker updated to reflect this project as ACTIVE if it is no longer on hold? Thanks!
https://stackoverflowteams.com/c/expensify/questions/12234

**Comment by johnmlee101** [comment]
I have read and reviewed this Design Doc!

**Comment by grgia** [comment]
working on manual tests and slightly updating the mocks to match our current components (no real updates to major flows, besides split)

**Comment by LLPeckham** [comment]
I have read and reviewed this Design Doc!

**Comment by grgia** [comment]
Discussion about including split started here - https://expensify.slack.com/archives/C036QM0SLJK/p1717492905833899

**Comment by grgia** [comment]
Catching up from having the flu, we should be able to move this forward next week

**Comment by BrtqKr** [comment]
@grgia are there any things that need to be resolved, or can we pass it for the review?

**Comment by grgia** [comment]
@BrtqKr I just need to update the manual tests section, I can get that done before EOW.

**Comment by grgia** [comment]
Working on a summer release project right now, aiming to update the final doc ASAP

**Comment by grgia** [comment]
I'm wrapping up Pending/Scanning then will focus on wrapping this up.

**Comment by grgia** [comment]
Almost wrapped on the above

**Comment by grgia** [comment]
Went through doc, ensured it was filled out- https://expensify.slack.com/archives/C21FRDWCV/p1724163075264519
Sending out for rereview to get this out and rolling cc @BrtqKr

**Comment by BrtqKr** [comment]
Sorry for the delay, I got sick on Monday. 

@grgia @AndrewGable 
Partially related to https://github.com/Expensify/App/issues/47811, but more about the group setup in general

I've merged main and it seems like the provided group ID is either incorrect (I've tried a couple of variants besides the one from the branch) or requires further changes or an access, possibly provisioning profile. A couple of things to point out here:
1. I do not see any app group containers in the settings and the naming pattern from the [slack thread](https://swmansion.slack.com/archives/C04878MDF34/p1724421579541059?thread_ts=1724247659.877149&cid=C04878MDF34) would also suggest that there might not be any configured ones available, since [according to the docs ](https://developer.apple.com/documentation/xcode/configuring-app-groups)
> A container ID must begin with `group.`
2. I'm getting errors when attempting to access a group

Also, it turns out that without a provisioning profile, I won't be able to select a target for the share extension as well as assign it to the provided group. So I would ask you to configure those things in the first place. 

I've moved some parts of the changes to [this branch](https://github.com/software-mansion-labs/expensify-app-fork/tree/brtqkr/wire-up-share-extension-ios) - it already contains a share extension template with some parts of the code for processing everything, so please just set up a group if it doesn't exist and include it inside of the share extension.

**Comment by AndrewGable** [comment]
Sorry @BrtqKr - Can you list out the changes you need exactly made in the Apple Developer console? I am not familiar with everything required to set up. Thanks!

**Comment by BrtqKr** [comment]
@AndrewGable, going step-by-step:

Firstly start from my branch, otherwise you won't be able to access the extension I've prepared

Configuring app group:
1. Go to NewExpensify => Signing & Capabilities 

Verify if the App Groups section is visible above the Associated domains 
<img width="229" alt="image" src="https://github.com/user-attachments/assets/73ceb26b-ac3a-45ef-9916-392cee148ddd">

if it is, then it means that the group exists and I need a provisioning profile to be able to access it and I won't even be able to list it otherwise.

If it isn't...
2. `+ Capability` => App group
3. The App groups section should appear above associated domains and the entitlements should be modified 
4. Afterwards, add a group `group.com.expensify.chat` in this section 

Apply App group to the share extension 
1. Select a target ShareViewController
<img width="165" alt="image" src="https://github.com/user-attachments/assets/3167fb34-8f05-4dc5-a975-c66b84129732">

2. Repeat the whole process from the previous point to include it in the same group - make sure you use the same name pattern. 

After that push the changes to my branch. I'm not sure if this would be enough for me to access it after the configuration, but I know that in my case the operation of adding the group won't even be finished without the profile.

**Comment by johnmlee101** [comment]
I have read and reviewed this Design Doc!

**Comment by nkuoch** [comment]
I have read and reviewed this Design Doc!

**Comment by grgia** [comment]
**Progress Update**

- @BrtqKr sent IOS set up instructions for the developer console [here](https://github.com/Expensify/App/issues/36613#issuecomment-2321469579), @AndrewGable could you take a look at those when you get a chance?
- We created 2 issues for tracking IOS and Android implementation:
    - IOS -  https://github.com/Expensify/App/issues/48788
    - Android - https://github.com/Expensify/App/issues/48789

**Comment by AndrewGable** [comment]
@grgia - You are an app deployer now, so feel free to make any changes you need for this project. Please ping me if you have questions. 

I believe we added App groups here: https://github.com/Expensify/App/pull/47738

**Comment by grgia** [comment]
Working with @BrtqKr to get the app permissions. I was able to update them locally but couldn't push to your branch

**Comment by BrtqKr** [comment]
Verified the permissions from the diff, seemed to be working, but I'll continue working on that in the next week. We've decided to split the work with @filip-solecki - he'll be working on the the android in parallel.

**Comment by grgia** [comment]
Both Android and IOS issues are in progress.

**Comment by grgia** [comment]
Still making steady progress

**Comment by grgia** [comment]
IOS and Android are both in progress

**Comment by grgia** [comment]
Same as above, draft PRs expected for this week

**Comment by grgia** [comment]
Android and IOS are in progress, still waiting on PRs

**Comment by grgia** [comment]
Same as above

**Comment by grgia** [comment]
Making monthly for holidays

**Comment by grgia** [comment]
Chatting about unblocking IOS build here - https://expensify.slack.com/archives/C01GTK53T8Q/p1736339255068649

**Comment by grgia** [comment]
PR nearing final review!

**Comment by grgia** [comment]
in progress, still held on provisioning issue

**Comment by mallenexpensify** [comment]
@grgia assigned myself as BZ in case you ever need help (and... to follow along)

**Comment by grgia** [comment]
Working on the provisioning profile locally. You can find the convo here: https://expensify.slack.com/archives/C03TQ48KC/p1738889505601349

**Comment by grgia** [comment]
ADHOC ND BUILD SUCCESS wooo! 

We still need to get profiles for production / hybrid finalized

**Comment by mallenexpensify** [comment]
No clue what next steps are, commenting to remove overdue

**Comment by grgia** [comment]
PR in final review

**Comment by grgia** [comment]
Main project merged!! Now in clean up phase

**Comment by grgia** [comment]
Clean up / polish is ongoing, but the feature is live so closing out this tracking issue

**Comment by trjExpensify** [comment]
@grgia (or whoever worked on project managing this project), I noticed the OP isn't in the typical design doc format, so I was wondering if this step got completed to add a suite of regression tests to TestRail: 

- [x] Add regression tests so that QA can test your feature with every deploy ([instructions](https://stackoverflowteams.com/c/expensify/questions/363))

**Comment by grgia** [comment]
@trjExpensify Will reopen to double check on that

**Comment by grgia** [comment]
Wrapped up with F1 doc, haven't gotten to this

**Comment by grgia** [comment]
yes it looks like there are native share tests
